crates/rust-analyzer/src/global_state.rs

```diff
@@ -69,6 +74,7 @@ pub(crate) struct GlobalState {
     pub(crate) config: Config,
     pub(crate) analysis_host: AnalysisHost,
     pub(crate) diagnostics: DiagnosticCollection,
+    pub(crate) completion_resolve_data: FxHashMap<usize, CompletionResolveData>,
```

Comment:
We shouldn't introduce state here since we may lose it if a request is cancelled etc. Everything needed to resolve the edits should be encapsulated in the fields of the `CompletionItem` (including it's `data` field which we can stuff anything into). Maybe what we should do is pass a structure with a tag (or maybe an enum of data) in the `data` field that either allows us to compute the additional text edits for this particular case or better yet specifies what kind of completion this. `handle_code_lens_resolve` might provide some inspiration.